### DISCUSSION TITLE
Map panel: Display GPS position history

### DISCUSSION THREAD

**Discussion by ysuurmeijer** [foxglove/community#630] [Q&A]
Currently the map panel only displays the latest GPS position when using it for ROSbag replay. It would be very valuable to add support for analyzing historic trajectories based on a LocationFix topic. 

- Version: v1.29.0
- OS: Ubuntu (docker)
- Data source: rosbridge
- ROS distro (if applicable):

**Current Experience**
GPS map plot only shows the latest GPS position when replaying data from a ROS bag over ROS bridge which makes it very difficult to analyse trajectories.

**Desired Experience**
Ability to view full trajectory from a live GPS topic. e.g. Add an option to keep 'historic' GPS points plotted on the map, either by setting a flag for the entire plot, or for instance by adding a parameter for each series to keep N historic points plotted on the map.

**Comment by jtbandes** [comment]
Duplicate of foxglove/studio#2797

**Comment by jtbandes** [comment]
I am reopening this issue since it occurs to me that foxglove/studio#2797 was about preloading data, whereas this issue is about live connections. We should keep a history of points for live connections (like we do for the Plot panel).

Internal ticket: FG-1829

**Comment by iftahnaf** [comment]
Any news on this issue?

**Comment by lars-nagel** [comment]
I would also love to hear if there are any news on this :)

**Comment by Gozgur** [comment]
Any news on this? I'm having the same problem too. 

Edit: I converted gps topic into foxglove.geojson to display gps position history  for now.

**Comment by jmm-slamcore** [comment]
Something like this could also be useful in the 3D panel to be able to visualise trajectories from e.g. pose topics (not necessarily GPS)

**Comment by defunctzombie** [comment]
Thanks for the feedback. The next version of the app will have the ability to display gps position history for live playback. You'll also be able to configure whether to display the history or to turn it off.
